apps/web/src/hooks/useSendCallsWithLogs.ts

```diff
@@ -0,0 +1,187 @@
+import { useAnalytics } from 'apps/web/contexts/Analytics';
+import { useErrors } from 'apps/web/contexts/Errors';
+import { decodeRawLog, USER_OPERATION_EVENT_LOG_NAME } from 'apps/web/src/utils/transactionLogs';
+import { ActionType } from 'libs/base-ui/utils/logEvent';
+import { useCallback, useEffect, useState } from 'react';
+import { Chain } from 'viem';
+import { SendCallsParameters } from 'viem/experimental';
+import { useAccount, useSwitchChain, useWaitForTransactionReceipt } from 'wagmi';
+import { useCallsStatus, useSendCalls } from 'wagmi/experimental';
+
+/*
+  A hook to request and track a wallet write transaction
+
+  Responsabilities:
+
+*/
+
+export enum BatchCallsStatus {
+  // Wallet request
+  Idle = 'idle',
+  Initiated = 'initiated',
+  Approved = 'approved',
+  Canceled = 'canceled',
+
+  // Top transaction
+  Processing = 'processing',
+  Reverted = 'reverted',
+
+  // UserOperationEvent and top transaction are successfull
+  Failed = 'failed',
+  Success = 'success',
+}
+
+export type UseSendCallsWithLogsProps = {
+  chain: Chain;
+  eventName: string;
+};
+
+export default function useSendCallsWithLogs({ chain, eventName }: UseSendCallsWithLogsProps) {
+  // Errors & Analytics
+  const { logEventWithContext } = useAnalytics();
+  const { logError } = useErrors();
+  const { connector } = useAccount();
+  const isCoinbaseSmartWallet = connector?.id === 'coinbaseWalletSDK';
+
+  const batchCallsEnabled = isCoinbaseSmartWallet;
+
+  const { chain: connectedChain } = useAccount();
+
+  const [batchCallsStatus, setBatchCallsStatus] = useState<BatchCallsStatus>(BatchCallsStatus.Idle);
+
+  // Experimental: Send a batch call
+  const {
+    sendCallsAsync,
+    data: sendCallsId,
+    isPending: sendCallsIsPending,
+    isSuccess: sendCallsIsSuccess,
+    isError: sendCallsIsError,
+    error: sendCallsError,
+  } = useSendCalls();
+
+  // Experimental: Track batch call status
+  const { data: sendCallsResult, isPending: sendCallsResultIsPending } = useCallsStatus({
+    // @ts-expect-error: query is disabled without a sendCallsId
```

Comment:
nit: maybe explain why the error is occurring instead of only why this is permissible here